source/vibe/utils/array.d

```diff
@@ -376,7 +376,7 @@ struct FixedRingBuffer(T, size_t N = 0, bool INITIALIZE = true) {
 		assert(r.m_buffer is m_buffer);
 		if( m_start + m_fill > m_buffer.length ){
 			assert(r.m_start >= m_start && r.m_start < m_buffer.length || r.m_start < mod(m_start+m_fill));
-			if( r.m_start > m_start ){
+			if( r.m_start >= m_start ){
```

Comment:
Would probably be a nice optimization to do `popFront` in case of `r.m_start == m_start`.